src/lib.rs

```diff
@@ -318,9 +320,45 @@ where
     /// ```
     pub fn update_display_buffer(&mut self, location: LedLocation, enabled: bool) {
         // TODO Validate `address` parameter.
-
         // Turn on/off the specified LED.
-        self.buffer[location.row_as_index()].set(location.common, enabled);
+        if enabled {
+            self.buffer[location.row as usize] |= location.common.bits();
+        } else {
+            self.buffer[location.row as usize] &= !location.common.bits();
+        }
+    }
+
+    /// Update the mask of an entire row.
+    ///
+    /// The buffer must be written using [write_display_buffer()](struct.HT16K33.html#method.write_display_buffer)
+    /// for the change to be displayed.
+    ///
+    /// # Arguments
+    ///
+    /// * `row` - The row to update.
+    /// * `mask` - Set the row to this mask.
+    /// ```
+    /// # use ht16k33::i2c_mock::I2cMock;
+    /// # use ht16k33::HT16K33;
+    /// # use ht16k33::ValidationError;
+    /// use ht16k33::{DisplayData, DisplayDataAddress};
+    /// # fn main() -> Result<(), ValidationError> {
+    /// # let mut i2c = I2cMock::new();
+    /// # let address = 0u8;
+    ///
+    /// let mut ht16k33 = HT16K33::new(i2c, address);
+    ///
+    /// ht16k33.update_row_mask(
+    ///     DisplayDataAddress::ROW_0 as usize,
+    ///     DisplayData::from_bits(0b1100_1100).unwrap().bits()
+    /// );
+    /// ht16k33.write_display_buffer();
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub fn update_row_mask(&mut self, row: usize, mask: u8) {
```

Comment:
This function was written before the buffer was marked public. Should it be removed? Should the buffer remain private?